packages/core/src/browser/keyboard/keyboard-browser.ts

```diff
@@ -0,0 +1,184 @@
+/********************************************************************************
+ * Copyright (C) 2019 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, postConstruct } from 'inversify';
+import { isOSX, isWindows } from '../../common/os';
+import { NativeKeyboardLayout, KeyboardLayoutProvider, KeyboardLayoutClient } from '../../common/keyboard/layout-provider';
+
+@injectable()
+export class BrowserKeyboardLayoutProvider implements KeyboardLayoutProvider {
+
+    private linuxFrench = require('../../../src/common/keyboard/layouts/linux-fr-French.json');
+    private linuxGerman = require('../../../src/common/keyboard/layouts/linux-de-German.json');
+    private macUS = require('../../../src/common/keyboard/layouts/mac-en-US.json');
+    private macFrench = require('../../../src/common/keyboard/layouts/mac-fr-French.json');
+    private macGerman = require('../../../src/common/keyboard/layouts/mac-de-German.json');
+    private winUS = require('../../../src/common/keyboard/layouts/win-en-US.json');
+    private winFrench = require('../../../src/common/keyboard/layouts/win-fr-French.json');
+    private winGerman = require('../../../src/common/keyboard/layouts/win-de-German.json');
+
+    protected get allLayouts(): NativeKeyboardLayout[] {
+        return [
+            this.winUS, this.macUS, this.winFrench, this.macFrench, this.winGerman, this.macGerman
+        ];
+    }
+
+    private client?: KeyboardLayoutClient;
+
+    setClient(client?: KeyboardLayoutClient): void {
```

Comment:
Why use this pattern over some event emitter able to handle multiple clients?

In this case there can only be one client for this provider (which is in singleton scope).

Reply:
Nevermind, just understood we had to interact with `KeyboardLayoutService` and not this class directly, my bad :)